app/models/user.rb

```diff
@@ -27,6 +27,20 @@ class User < ApplicationRecord
   scope :by_place,
         ->(place){where(country_id: place.presence || Settings.place.range)}
 
+  scope :by_name_like,
+        ->(name){where "name LIKE ?", "%#{name}%"}
+
+  scope :by_type_of,
+        ->(type_of){where(type_of: type_of.presence || Settings.type_of.range)}
```

Comment:
```suggestion
        ->(type_of){where(type_of: type_of if type_of.present?)}
```
Sử dụng như này đc ko em

Reply:
dạ em nghĩ là sử dụng như thế thì khi present? nó trả về theo giá trị nó truyền vào nếu true, còn cách của em là nếu nó ko nil thì sẽ lấy giá trị bản thân nó, còn nil thì sẽ lấy theo 1 array theo cách e xử lý.